scripts/upload_sample.py

```diff
@@ -0,0 +1,196 @@
+#! /usr/bin/env python
+"""Upload sample to Bonsai."""
+import click
+from functools import wraps
+import json
+import requests
+from requests.structures import CaseInsensitiveDict
+from pydantic import BaseModel, Field, FilePath
+from typing import Callable
+import yaml
+
+USER_ENV = "BONSAI_USER"
+PASSWD_ENV = "BONSAI_PASSWD"
+TIMEOUT = 10
+
+class SampleConfig(BaseModel):
+    """Definition of a sample configuration."""
+
+    group_id: str | None = Field(None, description="Add sample to group with id")
+    prp_result: FilePath = Field(..., description="Path to PRP output")
+    minhash_signature: FilePath = Field(..., description="Path to minhash signature")
+
+
+    def assinged_to_group(self) -> bool:
+        return self.group_id is not None
+
+
+class TokenObject(BaseModel):  # pylint: disable=too-few-public-methods
+    """Token object"""
+
+    token: str
+    type: str
+
+
+class ExecutionContext(BaseModel):
+    """Container for execution context."""
+
+    username: str
+    password: str
+    api_url: str
+    token: TokenObject | None = None
+
+
+def get_auth_token(ctx: ExecutionContext) -> TokenObject:
+    """Get authentication token from api"""
+    # configure header
+    headers = CaseInsensitiveDict()
+    headers["Content-Type"] = "application/x-www-form-urlencoded"
+    url = f'{ctx.api_url}/token'
+    resp = requests.post(
+        url,
+        data={"username": ctx.username, "password": ctx.password},
+        headers=headers,
+        timeout=TIMEOUT,
+    )
+    # controll that request
+    resp.raise_for_status()
+    json_res = resp.json()
+    token_obj = TokenObject(token=json_res["access_token"], type=json_res["token_type"])
+    return token_obj
+
+
+def api_authentication(func: Callable) -> Callable:
+    """Use authentication token for api.
+
+    :param func: API function to wrap with API auth headers
+    :type func: Callable
+    :return: Wrapped API function
+    :rtype: Callable
+    """
+
+    @wraps(func)
+    def wrapper(token_obj: TokenObject, *args, **kwargs) -> Callable:
+        """Add authentication headers to API requests.
+
+        :param token_obj: Auth token object
+        :type token_obj: TokenObject
+        :return: Wrapped API call function
+        :rtype: Callable
+        """
+        headers = CaseInsensitiveDict()
+        headers["Accept"] = "application/json"
+        headers["Authorization"] = f"{token_obj.type.capitalize()} {token_obj.token}"
+
+        return func(headers=headers, *args, **kwargs)
+
+    return wrapper
+
+
+@api_authentication
+def upload_sample(headers: CaseInsensitiveDict, ctx: ExecutionContext, cnf: SampleConfig) -> str:
+    """Create a new sample."""
+    sample_obj = json.load(cnf.prp_result.open())
+    resp = requests.post(f'{ctx.api_url}/samples', headers=headers, json=sample_obj, timeout=TIMEOUT)
+
+    resp.raise_for_status()
+    resp_data = resp.json()
+    return resp_data["sample_id"]
```

Comment:
This actually doesn't work because you removed `sample_id` from the output in [this PR](https://github.com/Clinical-Genomics-Lund/bonsai-prp/pull/47/files#r1530289877).